MyOnboarding/test/MyOnboardingApp.Api.Tests/TodoListControllerTest.cs

```diff
@@ -58,12 +88,27 @@ public async Task Put_IdSpecifiedTextSpecified_ReturnsOkStatusCode()
         [Test]
         public async Task Post_NewTextSpecifiedInRequestBody_ReturnsCorrectResponse()
         {
-            var msg = await _controller.GetMessageFromAction(controller => controller.PostAsync(new TodoListItem { Id = _expectedId, Text = "newText" }));
-            TodoListItem itemFromMsg;
-            msg.TryGetContentValue(out itemFromMsg);
+            var itemToAdd = new TodoListItem {Text = "newText"};
+            var expectedItem = new TodoListItem {Id = _expectedId, Text = "newText"};
+            _repository.AddNewItemAsync(itemToAdd).Returns(expectedItem);
+            const string expectedUrl = "expected/Url";
+            _itemUrlLocator.GetListItemUrl(Arg.Any<Guid>()).Returns(expectedUrl);
+
+            _controller.Configuration = new HttpConfiguration();
+            _controller.Configuration.Routes.MapHttpRoute(
+                name: "ListItemUrl",
+                routeTemplate: "api/v{version}/{controller}/{id}",
+                defaults: new {id = RouteParameter.Optional});
+
+            _controller.RequestContext.RouteData = new HttpRouteData(
+                route: new HttpRoute(),
+                values: new HttpRouteValueDictionary{{"version", "1.0"}, {"controller", "todolist"}});
```

Comment:
You should not need this here (since you introduced abstraction for routing), should you?